packages/components/src/form-token-field/index.tsx

```diff
@@ -169,10 +169,10 @@ export function FormTokenField( props: FormTokenFieldProps ) {
 	function onKeyDown( event: KeyboardEvent ) {
 		let preventDefault = false;
 
-		if ( event.defaultPrevented ) {
+		if ( event.defaultPrevented || event.nativeEvent.isComposing ) {
```

Comment:
Ok, we have another complication, sorry I didn't notice it the first time.

On Mac Safari, the <kbd>Enter</kbd> or <kbd>Backspace</kbd> that ends IME composition is `isComposing=false` 🙃 This causes these keydowns to immediately commit the token or erase the previous token, when that is not the user intent.


https://user-images.githubusercontent.com/555336/200918022-4eaa0342-ef5e-4f70-82d2-a9139cb0a8d4.mp4

<img width="175" alt="event output of the video" src="https://user-images.githubusercontent.com/555336/200918030-979dfb11-f340-4e00-a5ea-bc06d2135996.png">

☝️ `console.log( event.key, event.keyCode, event.nativeEvent.isComposing )`

At this point I'd consider reverting to the `event.keyCode` method, but maybe we can try a hybrid approach, given that `event.keyCode` could theoretically be removed from browsers at any time.

How about we add a `event.keyCode === 229` check in the guard? This way, even if `event.keyCode` is removed from browsers at some point, the damage will be limited to Safari.

```suggestion
		if (
			event.defaultPrevented ||
			// Ignore keydowns from IMEs
			event.nativeEvent.isComposing ||
			// Workaround for Mac Safari where the final Enter/Backspace of an IME composition
			// is `isComposing=false`, even though it's technically still part of the composition.
			// These can only be detected by keyCode.
			event.keyCode === 229
		) {
```

Reply:
TIL about IME composition 🤯 

Just skimming through the spec, is there any chance we could leverage `compositionstart` and `compositionend` events?

Reply:
Oh hi, welcome to cross-browser IME event handling hell!

> Just skimming through the spec, is there any chance we could leverage `compositionstart` and `compositionend` events?

Unfortunately that is a whole other can of worms I'd prefer we keep shut 😂 Japanese article, but [look at this table](https://qiita.com/pochman/items/b99c835bf598810d3c18#ime%E5%85%A5%E5%8A%9B%E3%81%AE%E7%B5%82%E4%BA%86enter%E3%81%AB%E3%82%88%E3%82%8B%E6%B1%BA%E5%AE%9A) of how random the event firing is across browsers. And yeah, the `Enter` keydown that we want to ignore in this case fires after `compositionend` in Safari, so it wouldn't help 🙃

Reply:
Thanks for the suggestion and the detailed research, @mirka !

I found [another article](https://developer.squareup.com/blog/understanding-composition-browser-events/) that corroborates the issue of the strange sequence of events in Safari.

> This is what the user should expect; only after finishing the composition should subsequent enter presses open their search query in a new tab. However, in my testing, Safari appears to fire the keyDown and compositionEnd events in the opposite order. Here’s the logic path: